include/TPP/Dialect/VNNI/VNNIOps.td

```diff
@@ -15,19 +20,25 @@ def VNNI_Dialect : Dialect {
   }];
 }
 
-def VNNI_MatmulOp : Op<VNNI_Dialect, "matmul"> {
-  let summary = [{Matmul operation with VNNI format of the first operand}];
+def VNNI_MatmulOp : Op<VNNI_Dialect, "matmul", [TypesMatchWith<"result type matches type of dest",
+                   "dest", "matrixC",
+                   "$_self">]> {
+  let summary = [{Matmul operation with VNNI format of the first operand with block
+                  size specified.}];
   let description = [{
-    Matmul Operation with VNNI format of the first operand..
+    Matmul Operation with VNNI format of the first operand.
+    For example, the operation performed is
+    C[i][j] += A[i/2][k][2]* B[k][j] with the first operand A's layout being
+    blocked by size 2.
 
     ```mlir
-    vnni.matmul(%arg0, %arg1): memref<2x4x2xbf16>, memref<4x6xbf16>, memref<4x6xbf16>
+    vnni.matmul(%arg0: memref<3x4x2xbf16>, %arg1: memref<4x6xbf16>)-> %arg2: memref<6x6xbf16>
```

Comment:
Looks still a bit strange to me. What about:
``` vnni.matmul ins(%arg0: memref<3x4x2xbf16>, %arg1: memref<4x6xbf16>) out(%arg2: memref<6x6xbf16>)```

Reply:
Also can you please add a test along the line of: https://github.com/plaidml/tpp-sandbox/blob/main/test/TPP/tpp/tpp-ops.mlir